### ISSUE TITLE
Jicofo port conflict after ejabberd update to 17.11

### ISSUE THREAD

**Issue by jlemangarin** [jitsi/jicofo#256]
Hello,

I'm facing an issue with jicofo, trying to update ejabberd from 16.12 to 17.11.
Ejabberd works fine after the update but jicofo seems to have a 'conflict' when it tries to connect to the focus service of ejabberd.

First thing first, let's verify that ejabberd is properly listening on this port (5348) before starting jicofo : 

> [root@gofast-comm-internal update]# netstat -tunlp
> tcp 0 0 0.0.0.0:5348 0.0.0.0:* LISTEN 27618/beam.smp
> [root@gofast-comm-internal update]# tail -n 500 /opt/ejabberd/logs/ejabberd.log
> 2017-12-01 17:45:52.178 [info] <0.354.0>@ejabberd_listener:init_tcp:140 Start accepting TCP connections at 0.0.0.0:5348 for ejabberd_service

 
Then, I start jicofo : 

> [root@gofast-comm-internal ~]# /usr/share/jicofo/jicofo.sh --domain=gofast-comm-internal.ceo-vision.com --secret=XXX --user_domain=gofast-comm-internal.ceo-vision.com --user_name=admin --user_password=XXX --subdomain=focus --host=localhost --port=5348 &> /var/log/jitsi/jicofo.log &
> [root@gofast-comm-internal update]# tail -f /opt/ejabberd/logs/ejabberd.log
> 2017-12-01 17:52:14.470 [info] <0.354.0>@ejabberd_listener:accept:302 (<0.547.0>) Accepted connection 127.0.0.1:47504 -> 127.0.0.1:5348
> 2017-12-01 17:52:14.728 [info] <0.546.0>@ejabberd_c2s:handle_auth_success:433 (tls|<0.545.0>) Accepted c2s PLAIN authentication for admin@gofast-comm-internal.ceo-vision.com by ldap backend from 127.0.0.1
> 2017-12-01 17:52:14.838 [info] <0.546.0>@ejabberd_c2s:bind:404 (tls|<0.545.0>) Opened c2s session for admin@gofast-comm-internal.ceo-vision.com/admin347759032950169
> 2017-12-01 17:52:19.554 [info] <0.354.0>@ejabberd_listener:accept:302 (<0.549.0>) Accepted connection 127.0.0.1:47505 -> 127.0.0.1:5348
> 2017-12-01 17:52:24.556 [info] <0.354.0>@ejabberd_listener:accept:302 (<0.551.0>) Accepted connection 127.0.0.1:47506 -> 127.0.0.1:5348

And jicofo throw this error:

> [root@gofast-comm-internal update]# tail -f /var/log/jitsi/jicofo.log
> déc. 01, 2017 5:55:19 PM net.java.sip.communicator.util.Logger error
> GRAVE: conflict, host:localhost, port:5348
> org.xmpp.component.ComponentException: conflict
> at org.jivesoftware.whack.ExternalComponent.connect(ExternalComponent.java:219)
> at org.jivesoftware.whack.ExternalComponentManager.addComponent(ExternalComponentManager.java:221)
> at org.jivesoftware.whack.ExternalComponentManager.addComponent(ExternalComponentManager.java:201)
> at org.jitsi.meet.ComponentMain$3.call(ComponentMain.java:270)
> at org.jitsi.meet.ComponentMain$3.call(ComponentMain.java:255)
> at org.jitsi.retry.RetryStrategy$TaskRunner.run(RetryStrategy.java:193)
> at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
> at java.util.concurrent.FutureTask.run(FutureTask.java:266)
> at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
> at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
> at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> at java.lang.Thread.run(Thread.java:748)

 
Do you have any clue about that ?
Thanks for your time !
Regards,

**Comment by jlemangarin** [comment]
Thanks a lot for your very fast answer !

I'm sure there is only one jicofo process running : 

> [root@localhost update]# ps -aux | grep jico
> root     28663 36.6  1.8 5205944 72244 pts/1   Sl   18:23   0:01 java -Xmx3072m -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/tmp -Djava.library.path=/usr/share/jicofo/lib/native/linux-64 -cp /usr/share/jicofo/jicofo.jar:/usr/share/jicofo/lib/agafua-syslog-0.4.jar:/usr/share/jicofo/lib/bccontrib-1.0.jar:/usr/share/jicofo/lib/bcpkix-jdk15on-1.54.jar:/usr/share/jicofo/lib/bcprov-jdk15on-1.54.jar:/usr/share/jicofo/lib/commons-codec-1.6.jar:/usr/share/jicofo/lib/commons-lang-2.4.jar:/usr/share/jicofo/lib/commons-lang3-3.1.jar:/usr/share/jicofo/lib/commons-logging-1.2.jar:/usr/share/jicofo/lib/concurrentlinkedhashmap-lru-1.0_jdk5.jar:/usr/share/jicofo/lib/core-2.0.0.jar:/usr/share/jicofo/lib/dnsjava-2.1.7.jar:/usr/share/jicofo/lib/dom4j-1.6.1.jar:/usr/share/jicofo/lib/fmj-1.0-20160121.172939-10.jar:/usr/share/jicofo/lib/guava-15.0.jar:/usr/share/jicofo/lib/httpclient-4.4.jar:/usr/share/jicofo/lib/httpcore-4.4.jar:/usr/share/jicofo/lib/ice4j-2.0-20170420.153542-14.jar:/usr/share/jicofo/lib/jain-sip-ri-ossonly-1.2.98c7f8c-jitsi-oss1.jar:/usr/share/jicofo/lib/java-sdp-nist-bridge-1.1.jar:/usr/share/jicofo/lib/javax.servlet-3.0.0.v201112011016.jar:/usr/share/jicofo/lib/javax.servlet-api-3.1.0.jar:/usr/share/jicofo/lib/jcip-annotations-1.0.jar:/usr/share/jicofo/lib/jetty-ajp-8.1.16.v20140903.jar:/usr/share/jicofo/lib/jetty-client-9.2.10.v20150310.jar:/usr/share/jicofo/lib/jetty-continuation-8.1.16.v20140903.jar:/usr/share/jicofo/lib/jetty-http-8.1.16.v20140903.jar:/usr/share/jicofo/lib/jetty-io-8.1.16.v20140903.jar:/usr/share/jicofo/lib/jetty-proxy-9.2.10.v20150310.jar:/usr/share/jicofo/lib/jetty-security-8.1.16.v20140903.jar:/usr/share/jicofo/lib/jetty-server-8.1.16.v20140903.jar:/usr/share/jicofo/lib/jetty-servlet-8.1.16.v20140903.jar:/usr/share/jicofo/lib/jetty-util-8.1.16.v20140903.jar:/usr/share/jicofo/lib/jetty-webapp-7.0.1.v20091125.jar:/usr/share/jicofo/lib/jetty-xml-7.0.1.v20091125.jar:/usr/share/jicofo/lib/jicoco-1.0-20161122.205032-37.jar:/usr/share/jicofo/lib/jitsi-android-osgi-1.0-20150723.002702-1.jar:/usr/share/jicofo/lib/jitsi-configuration-2.9-20150723.181638-1.jar:/usr/share/jicofo/lib/jitsi-credentialsstorage-2.9-20150723.181643-1.jar:/usr/share/jicofo/lib/jitsi-dnsservice-2.9-20150723.181644-1.jar:/usr/share/jicofo/lib/jitsi-netaddr-2.9-20150723.181645-1.jar:/usr/share/jicofo/lib/jitsi-protocol-2.9-20160823.174711-4.jar:/usr/share/jicofo/lib/jitsi-protocol-jabber-2.11-20170516.192102-5.jar:/usr/share/jicofo/lib/jitsi-protocol-media-2.9-20150723.181646-1.jar:/usr/share/jicofo/lib/jitsi-ui-service-2.9-20150723.181654-1.jar:/usr/share/jicofo/lib/jitsi-util-2.9-20160203.194039-2.jar:/usr/share/jicofo/lib/jna-4.1.0.jar:/usr/share/jicofo/lib/jnsapi-0.0.3-jitsi-1-20151013.145326-2.jar:/usr/share/jicofo/lib/json-simple-1.1.1.jar:/usr/share/jicofo/lib/jzlib-1.1.3.jar:/usr/share/jicofo/lib/libidn-1.15.jar:/usr/share/jicofo/lib/libjitsi-1.0-20170427.165204-276.jar:/usr/share/jicofo/lib/log4j-1.2.14.jar:/usr/share/jicofo/lib/orange-extensions-1.3.0.jar:/usr/share/jicofo/lib/org.apache.felix.framework-4.4.0.jar:/usr/share/jicofo/lib/org.apache.felix.main-4.4.0.jar:/usr/share/jicofo/lib/org.osgi.core-4.3.1.jar:/usr/share/jicofo/lib/sdes4j-1.1.3.jar:/usr/share/jicofo/lib/sdp-api-1.0.jar:/usr/share/jicofo/lib/slf4j-api-1.7.7.jar:/usr/share/jicofo/lib/slf4j-jdk14-1.7.7.jar:/usr/share/jicofo/lib/smack-3.2.2-jitsi-1-20151215.185921-3.jar:/usr/share/jicofo/lib/smackx-3.2.2-jitsi-1-20151215.185922-3.jar:/usr/share/jicofo/lib/tinder-1.2.3.jar:/usr/share/jicofo/lib/weupnp-0.1.4.jar:/usr/share/jicofo/lib/xml-apis-1.0.b2.jar:/usr/share/jicofo/lib/xmlpull-1.1.3.4a.jar:/usr/share/jicofo/lib/xpp3-1.1.4c.jar:/usr/share/jicofo/lib/zrtp4j-light-3.2.0-jitsi-1-20150723.002345-1.jar org.jitsi.jicofo.Main --domain=gofast-comm.ceo-vision.com --secret=XXXXX --user_domain=gofast-comm.ceo-vision.com --user_name=admin --user_password=XXXXX --subdomain=focus --host=localhost --port=5348
> root     28700  0.0  0.0 112664   952 pts/1    R+   18:23   0:00 grep --color=auto jico

**Comment by damencho** [comment]
Sorry, I'm not familiar with ejabberd, but prosody is using the same port for all component connections. And when I was seeing this error: "ComponentException: conflict",  this due to the fact that there is already jicofo instance using the same name as a component...

**Comment by jlemangarin** [comment]
Hello !

Sorry for the late answer, I don't have acces to this machine outside the office.

So, clearly, before starting Jicofo, nothing at all is doing registration request to ejabberd : 

```
2018-01-23 08:11:31.557 [info] <0.508.0>@eldap:connect_bind:1049 LDAP connection on gofast-dev.ceo-vision.com:636
2018-01-23 08:11:31.557 [info] <0.509.0>@eldap:connect_bind:1049 LDAP connection on gofast-dev.ceo-vision.com:636
2018-01-23 08:11:31.557 [info] <0.511.0>@eldap:connect_bind:1049 LDAP connection on gofast-dev.ceo-vision.com:636
2018-01-23 08:11:31.557 [info] <0.512.0>@eldap:connect_bind:1049 LDAP connection on gofast-dev.ceo-vision.com:636
2018-01-23 08:11:31.610 [info] <0.62.0>@ejabberd_cluster_mnesia:wait_for_sync:122 Waiting for Mnesia synchronization to complete
2018-01-23 08:11:31.610 [info] <0.62.0>@ejabberd_app:start:58 ejabberd 17.11 is started in the node ejabberd@localhost in 1.35s
2018-01-23 08:11:31.611 [info] <0.31.0> Application ejabberd started on node ejabberd@localhost
2018-01-23 08:11:31.611 [info] <0.355.0>@ejabberd_listener:init_tcp:140 Start accepting TCP connections at 0.0.0.0:5348 for ejabberd_service
2018-01-23 08:11:31.611 [info] <0.354.0>@ejabberd_listener:init_tcp:140 Start accepting TCP connections at 0.0.0.0:5347 for ejabberd_service
2018-01-23 08:11:31.611 [info] <0.353.0>@ejabberd_listener:init_tcp:140 Start accepting TCP connections at 0.0.0.0:5280 for ejabberd_http
2018-01-23 08:11:31.612 [info] <0.352.0>@ejabberd_listener:init_tcp:140 Start accepting TCP connections at 0.0.0.0:5275 for ejabberd_service
2018-01-23 08:11:31.612 [info] <0.351.0>@ejabberd_listener:init_tcp:140 Start accepting TCP connections at 0.0.0.0:5269 for ejabberd_s2s_in
2018-01-23 08:11:31.612 [info] <0.350.0>@ejabberd_listener:init_tcp:140 Start accepting TCP connections at 0.0.0.0:5222 for ejabberd_c2s
2018-01-23 08:11:31.612 [info] <0.346.0>@ejabberd_listener:init_udp:114 Start accepting UDP connections at 0.0.0.0:3478 for ejabberd_stun
2018-01-23 08:11:31.630 [info] <0.31.0> Application stun started on node ejabberd@localhost
```

At this point, 5348 is fully open to process incoming requests. Then I start Jicofo... And the very first request triggers the error.

Ejabberd's logs : 
```
2018-01-23 08:18:51.019 [info] <0.350.0>@ejabberd_listener:accept:302 (<0.544.0>) Accepted connection 127.0.0.1:45435 -> 127.0.0.1:5222
2018-01-23 08:18:51.144 [info] <0.355.0>@ejabberd_listener:accept:302 (<0.546.0>) Accepted connection 127.0.0.1:40698 -> 127.0.0.1:5348
2018-01-23 08:18:51.525 [info] <0.545.0>@ejabberd_c2s:handle_auth_success:433 (tls|<0.544.0>) Accepted c2s PLAIN authentication for admin@gofast-comm.ceo-vision.com by ldap backend from 127.0.0.1
2018-01-23 08:18:51.632 [info] <0.545.0>@ejabberd_c2s:bind:404 (tls|<0.544.0>) Opened c2s session for admin@gofast-comm.ceo-vision.com/admin52373773800511
```
And Jicofo's logs (timestamps are the same) : 

```
GRAVE: conflict, host:localhost, port:5348
org.xmpp.component.ComponentException: conflict
        at org.jivesoftware.whack.ExternalComponent.connect(ExternalComponent.java:219)
        at org.jivesoftware.whack.ExternalComponentManager.addComponent(ExternalComponentManager.java:221)
        at org.jivesoftware.whack.ExternalComponentManager.addComponent(ExternalComponentManager.java:201)
        at org.jitsi.meet.ComponentMain$3.call(ComponentMain.java:270)
        at org.jitsi.meet.ComponentMain$3.call(ComponentMain.java:255)
        at org.jitsi.retry.RetryStrategy$TaskRunner.run(RetryStrategy.java:193)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

janv. 23, 2018 8:18:51 AM net.java.sip.communicator.util.Logger info
INFOS: New component discovered: gofast-comm.ceo-vision.com, Version[ejabberd(17.11,unix/linux 3.10.0)@-2035748024
janv. 23, 2018 8:18:51 AM net.java.sip.communicator.util.Logger info
INFOS: Detected XMPP server version: Version[ejabberd(17.11,unix/linux 3.10.0)@-2035748024
janv. 23, 2018 8:18:51 AM net.java.sip.communicator.util.Logger info
INFOS: New component discovered: irc.gofast-comm.ceo-vision.com, null
janv. 23, 2018 8:18:51 AM net.java.sip.communicator.util.Logger info
INFOS: MUC component discovered: irc.gofast-comm.ceo-vision.com
janv. 23, 2018 8:18:51 AM net.java.sip.communicator.util.Logger info
INFOS: New component discovered: pubsub.gofast-comm.ceo-vision.com, null
janv. 23, 2018 8:18:51 AM net.java.sip.communicator.util.Logger info
INFOS: New component discovered: conference.gofast-comm.ceo-vision.com, null
janv. 23, 2018 8:18:51 AM net.java.sip.communicator.util.Logger info
INFOS: Services re-discovery interval: 30000
janv. 23, 2018 8:18:51 AM org.jitsi.util.LoggerImpl log
INFOS: XMPP provider reg state: RegistrationState=Registered
janv. 23, 2018 8:18:51 AM org.jitsi.util.LoggerImpl log
INFOS: XMPP provider Jabber:admin@gofast-comm.ceo-vision.com/admin52373773800511@localhost connected (JID: admin@gofast-comm.ceo-vision.com/admin52373773800511)
janv. 23, 2018 8:18:56 AM net.java.sip.communicator.util.Logger error
GRAVE: conflict, host:localhost, port:5348
org.xmpp.component.ComponentException: conflict
```
I'm really back to the wall, I can't find any explaination at this time.. :sob: 

Thanks again for your time !

**Comment by jlemangarin** [comment]
I tried to analyse the first exanches between Ejabberd and Jicofo. This is what I found : 

Jicofo tries to register : 
`<stream:stream.xmlns="jabber:component:accept".xmlns:stream="http://etherx.jabber.org/streams".to="focus.gofast-comm.ceo-vision.com">`

Ejabberd answers : 
`<stream stream.id='7830175151112861697'.xml:lang='en'.xmlns:stream='http://etherx.jabber.org/streams'.from='focus.gofast-comm.ceo-vision.com'.xmlns='jabber:component:accept'>`

And then, throws this error : 
```
<stream:error>
  <conflict.xmlns='urn:ietf:params:xml:ns:xmpp-streams'/>
  <text.xml:lang='en'.xmlns='urn:ietf:params:xml:ns:xmpp-streams'>
    Unable.to.register.route.on.existing.local.domain
  </text>
</stream:error>
```

Full dump : 

```
09:35:46.089090 IP localhost.42300 > localhost.5348: Flags [S], seq 2546602781, win 43690, options [mss 65495,sackOK,TS val 2068003 ecr 0,nop,wscale 8], length 0
        0x0000:  4500 003c 9d3e 4000 4006 9f7b 7f00 0001  E..<.>@.@..{....
        0x0010:  7f00 0001 a53c 14e4 97ca 131d 0000 0000  .....<..........
        0x0020:  a002 aaaa fe30 0000 0204 ffd7 0402 080a  .....0..........
        0x0030:  001f 8e23 0000 0000 0103 0308            ...#........
20:13:27.617683 IP localhost.5348 > localhost.42300: Flags [S.], seq 2314400541, ack 2546602782, win 43690, options [mss 65495,sackOK,TS val 2068003 ecr 2068003,nop,wscale 8], length 0
        0x0000:  4500 003c 0000 4000 4006 3cba 7f00 0001  E..<..@.@.<.....
        0x0010:  7f00 0001 14e4 a53c 89f2 f31d 97ca 131e  .......<........
        0x0020:  a012 aaaa fe30 0000 0204 ffd7 0402 080a  .....0..........
        0x0030:  001f 8e23 001f 8e23 0103 0308            ...#...#....
09:35:46.089125 IP localhost.42300 > localhost.5348: Flags [.], ack 1, win 171, options [nop,nop,TS val 2068003 ecr 2068003], length 0
        0x0000:  4500 0034 9d3f 4000 4006 9f82 7f00 0001  E..4.?@.@.......
        0x0010:  7f00 0001 a53c 14e4 97ca 131e 89f2 f31e  .....<..........
        0x0020:  8010 00ab fe28 0000 0101 080a 001f 8e23  .....(.........#
        0x0030:  001f 8e23                                ...#
09:35:46.094122 IP localhost.42300 > localhost.5348: Flags [P.], seq 1:134, ack 1, win 171, options [nop,nop,TS val 2068008 ecr 2068003], length 133
        0x0000:  4500 00b9 9d40 4000 4006 9efc 7f00 0001  E....@@.@.......
        0x0010:  7f00 0001 a53c 14e4 97ca 131e 89f2 f31e  .....<..........
        0x0020:  8018 00ab fead 0000 0101 080a 001f 8e28  ...............(
        0x0030:  001f 8e23 3c73 7472 6561 6d3a 7374 7265  ...#<stream:stre
        0x0040:  616d 2078 6d6c 6e73 3d22 6a61 6262 6572  am.xmlns="jabber
        0x0050:  3a63 6f6d 706f 6e65 6e74 3a61 6363 6570  :component:accep
        0x0060:  7422 2078 6d6c 6e73 3a73 7472 6561 6d3d  t".xmlns:stream=
        0x0070:  2268 7474 703a 2f2f 6574 6865 7278 2e6a  "http://etherx.j
        0x0080:  6162 6265 722e 6f72 672f 7374 7265 616d  abber.org/stream
        0x0090:  7322 2074 6f3d 2266 6f63 7573 2e67 6f66  s".to="focus.gof
        0x00a0:  6173 742d 636f 6d6d 2e63 656f 2d76 6973  ast-comm.ceo-vis
        0x00b0:  696f 6e2e 636f 6d22 3e                   ion.com">
09:35:46.094149 IP localhost.5348 > localhost.42300: Flags [.], ack 134, win 175, options [nop,nop,TS val 2068008 ecr 2068008], length 0
        0x0000:  4500 0034 7d3d 4000 4006 bf84 7f00 0001  E..4}=@.@.......
        0x0010:  7f00 0001 14e4 a53c 89f2 f31e 97ca 13a3  .......<........
        0x0020:  8010 00af fe28 0000 0101 080a 001f 8e28  .....(.........(
        0x0030:  001f 8e28                                ...(
09:35:46.094483 IP localhost.5348 > localhost.42300: Flags [P.], seq 1:196, ack 134, win 175, options [nop,nop,TS val 2068009 ecr 2068008], length 195
        0x0000:  4500 00f7 7d3e 4000 4006 bec0 7f00 0001  E...}>@.@.......
        0x0010:  7f00 0001 14e4 a53c 89f2 f31e 97ca 13a3  .......<........
        0x0020:  8018 00af feeb 0000 0101 080a 001f 8e29  ...............)
        0x0030:  001f 8e28 3c3f 786d 6c20 7665 7273 696f  ...(<?xml.versio
        0x0040:  6e3d 2731 2e30 273f 3e3c 7374 7265 616d  n='1.0'?><stream
        0x0050:  3a73 7472 6561 6d20 6964 3d27 3738 3330  :stream.id='7830
        0x0060:  3137 3531 3531 3131 3238 3631 3639 3727  175151112861697'
        0x0070:  2078 6d6c 3a6c 616e 673d 2765 6e27 2078  .xml:lang='en'.x
        0x0080:  6d6c 6e73 3a73 7472 6561 6d3d 2768 7474  mlns:stream='htt
        0x0090:  703a 2f2f 6574 6865 7278 2e6a 6162 6265  p://etherx.jabbe
        0x00a0:  722e 6f72 672f 7374 7265 616d 7327 2066  r.org/streams'.f
        0x00b0:  726f 6d3d 2766 6f63 7573 2e67 6f66 6173  rom='focus.gofas
        0x00c0:  742d 636f 6d6d 2e63 656f 2d76 6973 696f  t-comm.ceo-visio
        0x00d0:  6e2e 636f 6d27 2078 6d6c 6e73 3d27 6a61  n.com'.xmlns='ja
        0x00e0:  6262 6572 3a63 6f6d 706f 6e65 6e74 3a61  bber:component:a
        0x00f0:  6363 6570 7427 3e                        ccept'>
09:35:46.094542 IP localhost.5348 > localhost.42300: Flags [P.], seq 196:400, ack 134, win 175, options [nop,nop,TS val 2068009 ecr 2068008], length 204
        0x0000:  4500 0100 7d3f 4000 4006 beb6 7f00 0001  E...}?@.@.......
        0x0010:  7f00 0001 14e4 a53c 89f2 f3e1 97ca 13a3  .......<........
        0x0020:  8018 00af fef4 0000 0101 080a 001f 8e29  ...............)
        0x0030:  001f 8e28 3c73 7472 6561 6d3a 6572 726f  ...(<stream:erro
        0x0040:  723e 3c63 6f6e 666c 6963 7420 786d 6c6e  r><conflict.xmln
        0x0050:  733d 2775 726e 3a69 6574 663a 7061 7261  s='urn:ietf:para
        0x0060:  6d73 3a78 6d6c 3a6e 733a 786d 7070 2d73  ms:xml:ns:xmpp-s
        0x0070:  7472 6561 6d73 272f 3e3c 7465 7874 2078  treams'/><text.x
        0x0080:  6d6c 3a6c 616e 673d 2765 6e27 2078 6d6c  ml:lang='en'.xml
        0x0090:  6e73 3d27 7572 6e3a 6965 7466 3a70 6172  ns='urn:ietf:par
        0x00a0:  616d 733a 786d 6c3a 6e73 3a78 6d70 702d  ams:xml:ns:xmpp-
        0x00b0:  7374 7265 616d 7327 3e55 6e61 626c 6520  streams'>Unable.
        0x00c0:  746f 2072 6567 6973 7465 7220 726f 7574  to.register.rout
        0x00d0:  6520 6f6e 2065 7869 7374 696e 6720 6c6f  e.on.existing.lo
        0x00e0:  6361 6c20 646f 6d61 696e 3c2f 7465 7874  cal.domain</text
        0x00f0:  3e3c 2f73 7472 6561 6d3a 6572 726f 723e  ></stream:error>
09:35:46.094587 IP localhost.5348 > localhost.42300: Flags [P.], seq 400:416, ack 134, win 175, options [nop,nop,TS val 2068009 ecr 2068008], length 16
        0x0000:  4500 0044 7d40 4000 4006 bf71 7f00 0001  E..D}@@.@..q....
        0x0010:  7f00 0001 14e4 a53c 89f2 f4ad 97ca 13a3  .......<........
        0x0020:  8018 00af fe38 0000 0101 080a 001f 8e29  .....8.........)
        0x0030:  001f 8e28 3c2f 7374 7265 616d 3a73 7472  ...(</stream:str
        0x0040:  6561 6d3e                                eam>
09:35:46.094652 IP localhost.5348 > localhost.42300: Flags [F.], seq 416, ack 134, win 175, options [nop,nop,TS val 2068009 ecr 2068008], length 0
        0x0000:  4500 0034 7d41 4000 4006 bf80 7f00 0001  E..4}A@.@.......
        0x0010:  7f00 0001 14e4 a53c 89f2 f4bd 97ca 13a3  .......<........
        0x0020:  8011 00af fe28 0000 0101 080a 001f 8e29  .....(.........)
        0x0030:  001f 8e28                                ...(
09:35:46.106442 IP localhost.5348 > localhost.42300: Flags [F.], seq 416, ack 134, win 175, options [nop,nop,TS val 2068021 ecr 2068008], length 0
        0x0000:  4500 0034 7d42 4000 4006 bf7f 7f00 0001  E..4}B@.@.......
        0x0010:  7f00 0001 14e4 a53c 89f2 f4bd 97ca 13a3  .......<........
        0x0020:  8011 00af fe28 0000 0101 080a 001f 8e35  .....(.........5
        0x0030:  001f 8e28                                ...(
09:35:46.119044 IP localhost.42300 > localhost.5348: Flags [.], ack 196, win 175, options [nop,nop,TS val 2068031 ecr 2068009], length 0
        0x0000:  4500 0034 9d41 4000 4006 9f80 7f00 0001  E..4.A@.@.......
        0x0010:  7f00 0001 a53c 14e4 97ca 13a3 89f2 f3e1  .....<..........
        0x0020:  8010 00af fe28 0000 0101 080a 001f 8e3f  .....(.........?
        0x0030:  001f 8e29                                ...)
09:35:46.119062 IP localhost.42300 > localhost.5348: Flags [.], ack 400, win 180, options [nop,nop,TS val 2068031 ecr 2068009], length 0
        0x0000:  4500 0034 9d42 4000 4006 9f7f 7f00 0001  E..4.B@.@.......
        0x0010:  7f00 0001 a53c 14e4 97ca 13a3 89f2 f4ad  .....<..........
        0x0020:  8010 00b4 fe28 0000 0101 080a 001f 8e3f  .....(.........?
        0x0030:  001f 8e29                                ...)
09:35:46.119067 IP localhost.42300 > localhost.5348: Flags [.], ack 416, win 180, options [nop,nop,TS val 2068031 ecr 2068009], length 0
        0x0000:  4500 0034 9d43 4000 4006 9f7e 7f00 0001  E..4.C@.@..~....
        0x0010:  7f00 0001 a53c 14e4 97ca 13a3 89f2 f4bd  .....<..........
        0x0020:  8010 00b4 fe28 0000 0101 080a 001f 8e3f  .....(.........?
        0x0030:  001f 8e29                                ...)
09:35:46.119073 IP localhost.42300 > localhost.5348: Flags [.], ack 417, win 180, options [nop,nop,TS val 2068031 ecr 2068021,nop,nop,sack 1 {416:417}], length 0
        0x0000:  4500 0040 9d44 4000 4006 9f71 7f00 0001  E..@.D@.@..q....
        0x0010:  7f00 0001 a53c 14e4 97ca 13a3 89f2 f4be  .....<..........
        0x0020:  b010 00b4 fe34 0000 0101 080a 001f 8e3f  .....4.........?
        0x0030:  001f 8e35 0101 050a 89f2 f4bd 89f2 f4be  ...5............
09:35:46.127750 IP localhost.42300 > localhost.5348: Flags [P.], seq 134:197, ack 417, win 180, options [nop,nop,TS val 2068042 ecr 2068021], length 63
        0x0000:  4500 0073 9d45 4000 4006 9f3d 7f00 0001  E..s.E@.@..=....
        0x0010:  7f00 0001 a53c 14e4 97ca 13a3 89f2 f4be  .....<..........
        0x0020:  8018 00b4 fe67 0000 0101 080a 001f 8e4a  .....g.........J
        0x0030:  001f 8e35 3c68 616e 6473 6861 6b65 3e64  ...5<handshake>d
        0x0040:  3165 3536 3539 3466 3231 3063 6533 6539  1e56594f210ce3e9
        0x0050:  3563 3666 3238 3539 3732 6662 6264 3039  5c6f285972fbbd09
        0x0060:  3234 3161 6130 323c 2f68 616e 6473 6861  241aa02</handsha
        0x0070:  6b65 3e                                  ke>
09:35:46.127770 IP localhost.5348 > localhost.42300: Flags [R], seq 2314400958, win 0, length 0
        0x0000:  4500 0028 9362 4000 4006 a96b 7f00 0001  E..(.b@.@..k....
        0x0010:  7f00 0001 14e4 a53c 89f2 f4be 0000 0000  .......<........
        0x0020:  5004 0000 790c 0000                      P...y...
```

Is this helping ? Thanks !

**Comment by damencho** [comment]
I see this: https://github.com/processone/ejabberd/blob/master/src/ejabberd_service.erl#L120
Apparently, ejabberd does not recognize focus.gofast-comm.ceo-vision.com as a configured component and returns conflict ... in this situation, prosody returns error host-unknown which is more descriptive for the error.
Check your config ... does ejabberd has some admin interface, telnet/web something to check what it has configured...

**Comment by jlemangarin** [comment]
Hello again !

It seems like it's an ejabberd issue with my certificates. Ejabberd says there is no certificate matching "focus.gofast-comm-internal.ceo-vision.com", I guess that's the reason why it won't let jicofo connect.

I'm not already sure that's the problem but it's a very good track. in that case, it's not a Jicofo issue and I will have to open an ejabberd issue.

Thank you for your great help !

**Comment by jlemangarin** [comment]
Hello,

I resolved all certificate issues in ejabberd's logs (and all error messages) except these ones : 

```
2017-11-11 09:58:22.191 [warning] <0.65.0>@ejabberd_config:emit_deprecation_warning:1373 Module mod_http_bind is deprecated, use mod_bosh instead
2017-11-11 09:58:25.416 [warning] <0.110.0> More than 80% of OS memory is allocated, starting OOM watchdog
2017-11-11 09:58:25.963 [error] <0.389.0> backend port not found: inotifywait
```
I also checked all the ejabberd configuration in the admin interface, but nothing seems bad to me. Tell me if you want an acces to it, I can grant you one.

May have you another idea ? :)

Thank you !!

**Comment by damencho** [comment]
Sorry, I'm not familiar with ejabberd and will not be of any help. 
Why did you choose to not run it with the default prosody?

**Comment by jlemangarin** [comment]
Hello !

Since this issue is still opened, I guess my case remains a mistery. :)

I tried to upgrade to ejabberd 18.04 but the issue is still here. Jicofo is unable to register to ejabberd. 

So we're stucked to ejabberd 16.12 and it seems like new versions of jicofo are not super stable with this old ejabberd version ("Something wrong happened" message etc.)

Have you listed an issue like that these last months ?

Thank you !

**Comment by damencho** [comment]
Have you checked what is the problem when you see that message? Check the js console of the client for some errors.

**Comment by Neustradamus** [comment]
@jlemangarin: Any news on it?

**Comment by jlemangarin** [comment]
Yep, running Jicofo and jitsi-videobridge as services (create one if needed) resolves the issue.

Also jitsi-videobridge is unstable with ejabberd (it works but makes some strange unwanted errors and behaviors)

**Comment by Neustradamus** [comment]
@jlemangarin: What are the problems exactly?
Can you publish more informations about the error etc?
Ejabberd 20.xx?

**Comment by jlemangarin** [comment]
I don't really have more informations, starting Jicofo and Jitsi-videobridge as services instead of using the .sh scripts resolved the issue of this topic. 

But currently we moved to Prodosy, using Ejabberd we had a lot of videobridge errors and instabilities like :
SEVERE: Cannot find socket to remove.
WARNING: SCTP connection with 56856418 not ready yet.
WARNING: No available transport channel, can't send a message
INFO: Failed to handle TCP socket Socket
...

We had to wait like 30 or 40 seconds before the video/audio stream is connected and had instabilities (deconnections, video stream cuts..)
